Sources/BrowserServicesKit/SecureVault/AutofillSecureVault.swift

```diff
@@ -59,39 +73,31 @@ public protocol SecureVault {
     @discardableResult
     func storeCreditCard(_ card: SecureVaultModels.CreditCard) throws -> Int64
     func deleteCreditCardFor(cardId: Int64) throws
-}
-
-/// Protocols can't be nested, but classes can.  This struct provides a 'namespace' for the default implementations of the providers to keep it clean for other things going on in this library.
-internal struct SecureVaultProviders {
-
-    var crypto: SecureVaultCryptoProvider
-    var database: SecureVaultDatabaseProvider
-    var keystore: SecureVaultKeyStoreProvider
 
 }
 
-class DefaultSecureVault: SecureVault {
+public class DefaultAutofillSecureVault<T: AutofillDatabaseProvider>: AutofillSecureVault {
+
+    public typealias AutofillStorageProviders = SecureStorageProviders<T>
 
     private let lock = NSLock()
-    private let queue = DispatchQueue(label: "Secure Vault")
 
-    private let providers: SecureVaultProviders
+    private let providers: AutofillStorageProviders
     private let expiringPassword: ExpiringValue<Data>
 
-    var authExpiry: TimeInterval {
+    public var authExpiry: TimeInterval {
         return expiringPassword.expiresAfter
     }
 
-    internal init(authExpiry: TimeInterval,
-                  providers: SecureVaultProviders) {
+    public required init(providers: AutofillStorageProviders) {
         self.providers = providers
-        self.expiringPassword = ExpiringValue(expiresAfter: authExpiry)
+        self.expiringPassword = ExpiringValue(expiresAfter: 60 * 60 * 24 * 3)
```

Comment:
This does feel as though the constructor here should support additional parameters specific to the subclass, but I can live with this for now given our discussions about this stuff.